### PR TITLE
roaring: add version 0.5.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **roaring/0.5.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/roaring/all/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.1)
 project(cmake_wrapper)
 
-include("conanbuildinfo.cmake")
-conan_basic_setup()
+include(conanbuildinfo.cmake)
+conan_basic_setup(KEEP_RPATHS)
```
Comment (SpaceIm) [inline recipes/roaring/all/CMakeLists.txt:5]: Due to https://github.com/RoaringBitmap/CRoaring/blob/v0.5.0/CMakeLists.txt#L1 + https://github.com/RoaringBitmap/CRoaring/blob/v0.5.0/CMakeLists.txt#L8, `KEEP_RPATHS` is not sufficient.

Comment (toge) [reply recipes/roaring/all/CMakeLists.txt]: @SpaceIm 
Thank you.
Reverted it.

Comment (SpaceIm) [reply recipes/roaring/all/CMakeLists.txt]: You can keep it, I just say that more work is needed.

Comment (toge) [reply recipes/roaring/all/CMakeLists.txt]: Oh. Now I understand.
I will try to fix it.

**Review by SpaceIm — APPROVED** [review]

**Review by madebr — DISMISSED** [review]

File: recipes/roaring/all/CMakeLists.txt (hunk 1 above)
Comment (toge) [reply recipes/roaring/all/CMakeLists.txt:5]: @SpaceIm 
Sorry for late.
I fixed rpaths issue. Would you review it?
I am going to create PR in upstream repository.

File: recipes/roaring/all/conanfile.py [hunk 2]
```diff
@@ -66,19 +61,22 @@ def source(self):
...
+        cmake.definitions["ROARING_DISABLE_NATIVE"] = not self.options.native_optimization
+        cmake.definitions["ROARING_BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["ENABLE_ROARING_TESTS"] = False
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
 
     def build(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "cmake_minimum_required(VERSION 2.8)",
+            "cmake_minimum_required(VERSION 3.1)")
```
Comment (SpaceIm) [inline recipes/roaring/all/conanfile.py:78]:
```suggestion
```

File: recipes/roaring/all/conanfile.py [hunk 3]
```diff
@@ -66,19 +61,22 @@ def source(self):
...
-        self._cmake.definitions["ROARING_BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["ENABLE_ROARING_TESTS"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["ROARING_DISABLE_AVX"] = not self.options.get_safe("with_avx", False)
+        cmake.definitions["ROARING_DISABLE_NEON"] = not self.options.get_safe("with_neon", False)
+        cmake.definitions["ROARING_DISABLE_NATIVE"] = not self.options.native_optimization
+        cmake.definitions["ROARING_BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["ENABLE_ROARING_TESTS"] = False
```
Comment (SpaceIm) [inline recipes/roaring/all/conanfile.py:71]:
```suggestion
        cmake.definitions["ENABLE_ROARING_TESTS"] = False
        # Relocatable shared lib on Macos
        cmake.definitions["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
No need to change min CMake version

**Comment by toge** [comment]
@SpaceIm 
Thank you!

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**